hudi-common/src/main/java/org/apache/hudi/common/util/SpillableMapUtils.java

```diff
@@ -130,8 +130,11 @@ public static <R> R convertToHoodieRecordPayload(GenericRecord record, String pa
                                                    boolean withOperationField,
                                                    Option<String> partitionName) {
     final String recKey = record.get(recordKeyPartitionPathFieldPair.getKey()).toString();
-    final String partitionPath = (partitionName.isPresent() ? partitionName.get() :
-        record.get(recordKeyPartitionPathFieldPair.getRight()).toString());
+    String partitionPath = "";
+    if (recordKeyPartitionPathFieldPair.getRight() != null) {
+      partitionPath = (partitionName.isPresent() ? partitionName.get() :
+          record.get(recordKeyPartitionPathFieldPair.getRight()).toString());
```

Comment:
is this a separate bug you're fixing here? maybe a separate PR for this so it can be easily cherry-picked for the upcoming minor release

Reply:
Sorry, these code is unnecessary. And I will remove them.